### PR TITLE
[SYSTEMML-1468] [SYSTEMML-1463] New 1D/2D "Scale & Shift" `nn` layers + batch norm renaming

### PR DESCRIPTION
**Pull request by dusenberrymw**
This PR does the following:

1. Renames `batch_norm.dml` and `spatial_batch_norm.dml` to `batch_norm1d.dml` and `batch_norm2d.dml`.
2. Introduces new 1D/2D "Scale & Shift" layers: `scale_shift1d.dml`, `scale_shift2d.dml`.

A "Scale & Shift" layer introduces learnable parameters (`gamma`, `beta`) to scale and shift the input on either a per-feature basis (1D) or a per-channel basis (2D): `y = x*gamma + beta`.

### CODE REVIEW

**Comment by dusenberrymw** [comment]
cc @niketanpansare

**Comment by niketanpansare** [comment]
LGTM. Thanks @dusenberrymw
